changelog/changelog.py

```diff
@@ -167,18 +167,18 @@ def default_changelog_generator(item):
     return f" - {title}. {compat_msg} [View pull request]({pr_url})\n"
 
 
+def get_sct_function_from_label(labels=[]):
+    labels_list = []
+    for label in labels:
+        if "sct_" in label['name']:
+            labels_list.append(label['name'])
+    return labels_list
+
+
```

Comment:
Ah, and this whole comment is made moot by 78bebde86f6de9f68b8abb0f158ca0d000f77868 anyway.

Reply:
> I hadn't thought about the "redefining the function" aspect. Could you elaborate a little more on what you mean by that?

Sure! I think other languages do it differently, but in Python at least, a function definition is a statement like any other, and executing it has the effect of creating a new function object, and assigning the result to the given function name (which is just like any other variable name, and can be re-assigned, etc.).

The benefit is that you can define functions anywhere, including inside other functions, inside a loop, etc. This is very useful when you want a nested function to refer to some values that are local variables in some outer function. In particular, this is how function decorators are often implemented.

The minor downside is that the function doesn't get created just once (like a compiler would do it), but every time the `def` statement is reached, and like any other action, it takes a bit of time to execute.

(As a side note, the `class` statement is similarly just like any other statement, and it can appear inside a function, another class, a loop, etc. Part of how it works is that it executes the class body, which can contain any statement, not just function definitions!)

> _(It was my impression that nesting functions like this is meant to signal intent that "this function is a small utility function that is only meant to be used in the context of its outer function". I figured it was an organizational tool? I didn't realize it came with downsides! sweat_smile)_

It can certainly be used that way! But because of the (admittedly small) performance cost, I think it's unusual to do it when it happens in a loop.